samples/bluetooth/per_sync/src/main.c

```diff
@@ -0,0 +1,287 @@
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+
+	printk("PER_ADV_SYNC[%u]: [DEVICE]: %s sync terminated\n",
+	       bt_le_per_adv_sync_get_index(sync), le_addr);
+
+	k_sem_give(&sem_per_sync_lost);
+}
+
+static void recv_cb(struct bt_le_per_adv_sync *sync,
+		    const struct bt_le_per_adv_sync_recv_info *info,
+		    struct net_buf_simple *buf)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+
+	printk("PER_ADV_SYNC[%u]: [DEVICE]: %s, tx_power %i, "
+	       "RSSI %i, CTE %u, data length %u\n",
+	       bt_le_per_adv_sync_get_index(sync), le_addr, info->tx_power,
+	       info->rssi, info->cte_type, buf->len);
+}
+
+static struct bt_le_per_adv_sync_cb sync_callbacks = {
+	.synced = sync_cb,
+	.term = term_cb,
+	.recv = recv_cb
+};
+
+void main(void)
+{
+	struct bt_le_per_adv_sync_param sync_create_param;
+	struct bt_le_per_adv_sync *sync;
+	int err;
+
+	printk("Starting Periodic Advertising Synchronization Demo\n");
+
+#if defined(HAS_LED)
+	printk("Get reference to LED device...");
+	dev = device_get_binding(LED0);
+	if (!dev) {
+		printk("Failed.\n");
+		return;
+	}
+	printk("done.\n");
+
+	printk("Configure GPIO pin...");
+	err = gpio_pin_configure(dev, PIN, GPIO_OUTPUT_ACTIVE | FLAGS);
+	if (err) {
+		return;
+	}
+	printk("done.\n");
+
+	k_delayed_work_init(&blink_work, blink_timeout);
+#endif /* HAS_LED */
+
+	/* Initialize the Bluetooth Subsystem */
+	err = bt_enable(NULL);
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return;
+	}
+
+	printk("Scan callbacks register...");
+	bt_le_scan_cb_register(&scan_callbacks);
+	printk("success.\n");
+
+	printk("Periodic Advertising callbacks register...");
+	bt_le_per_adv_sync_cb_register(&sync_callbacks);
+	printk("Success.\n");
+
+	printk("Initialize semaphore sem_per_adv...");
+	err = k_sem_init(&sem_per_adv, 0, 1);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("success.\n");
+
+	printk("Initialize semaphore sem_per_sync...");
+	err = k_sem_init(&sem_per_sync, 0, 1);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("success.\n");
+
+	printk("Initialize semaphore sem_per_sync_lost...");
+	err = k_sem_init(&sem_per_sync_lost, 0, 1);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("success.\n");
+
+	printk("Start scanning...");
+	err = bt_le_scan_start(BT_LE_SCAN_ACTIVE, NULL);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("success.\n");
+
+	do {
+#if defined(HAS_LED)
+		printk("Start blinking LED...\n");
+		led_is_on = false;
+		gpio_pin_set(dev, PIN, (int)led_is_on);
+		k_delayed_work_submit(&blink_work, BLINK_ONOFF);
+#endif /* HAS_LED */
+
+		printk("Waiting for periodic advertising...\n");
+		is_per_adv = false;
+		err = k_sem_take(&sem_per_adv, K_FOREVER);
+		if (err) {
+			printk("failed (err %d)\n", err);
+			return;
+		}
+		printk("Found periodic advertising.\n");
+
```

Comment:
Did you consider stopping the scanner after the sync was established? 
I find it hard to see the prints related to the periodic advertiser in terminal output. It is overflowed by advertising reports from the extended advertiser.

Reply:
No, so as to be able to demo that periodic sync continues even when local events could be skipped due to scan events or possible extended advertising collisions on auxiliary channels from other devices.

But yes, if this is a sample to demo low power usage of periodic adv sync, scanning be disabled after sync established.